.github/workflows/build-test-pr.yml

```diff
@@ -64,110 +58,68 @@ jobs:
               - '.github/workflows/build-test-pr.yml'
               - 'extensions/msal-node-extensions/**/**.!(md)'
 
-  lib-build-test:
+  ci-msal-common:
+    name: msal-common
     needs: pre-check
-    runs-on: ubuntu-latest
-
-    strategy:
-        fail-fast: false
-        matrix:
-          # Parse JSON array containing names of all filters matching any of changed files
-          # e.g. ['package1', 'package2'] if both package folders contains changes
-          library: ${{ fromJSON(needs.pre-check.outputs.packages) }}
-
-    name: ${{ matrix.library }}
-
-    steps:
-    - uses: actions/checkout@v3
-
-    - name: Use Node.js
-      uses: actions/setup-node@v3
-      with:
-        node-version: 20
-
-    - name: Restore node_modules
-      uses: actions/cache@v3
-      id: cache
-      with:
-        path: |
-          node_modules
-          lib/*/node_modules
-        key: ${{ runner.os }}-${{ hashFiles('package-lock.json', 'lib/*/package-lock.json') }}
-
-    - name: Clean Install
-      if: steps.cache.outputs.cache-hit != 'true'
-      env:
-        RUNNING_NODE_CI: 1
-      run: npm ci
-
-    - name: Upload npm logs
-      uses: actions/upload-artifact@v3
-      if: failure()
-      with:
-        name: npm_logs_${{ matrix.library }}
-        path: ~/.npm/_logs/**/*
-        retention-days: 2
-
-    - name: Build packages
-      working-directory: lib/${{ matrix.library }}
-      run: npm run build:all
-
-    - name: Lint
-      working-directory: lib/${{ matrix.library }}
-      run: npm run lint
-
-    - name: Unit Tests with coverage
-      working-directory: lib/${{ matrix.library }}
-      run: npm run test:coverage
-
-    - name: Upload Test Coverage to CodeCov
-      if: success()
-      uses: codecov/codecov-action@v3
-      with:
-        files: lib/${{matrix.library}}/coverage/lcov.info
-        flags: ${{ matrix.library }}
-
-  extensions-build-test:
+    if: |
+      !failure() && 
+      contains(toJSON(needs.pre-publish.outputs.packages), 'common')
+    uses: AzureAD/microsoft-authentication-library-for-js/.github/workflows/ci-template.yml@update-ci-workflow-triggers
```

Comment:
branch name at the end is temporary to make sure workflow is working. Will update to dev before merging